src/main.ts

```diff
@@ -295,6 +317,24 @@ function handleKeyDown({ key }: KeyboardEvent): void {
   }
 }
 
+async function handleFindTrack(): Promise<void> {
+  // Load track value
+  const trackId = trackInput.valueAsNumber;
+  const newTrack = dataset!.buildTrack(trackId);
+
+  if (newTrack.length() < 1) {  // Check track validity
+    return;
+  }
+  selectedTrack = newTrack;
+  timeControls.setCurrentFrame(selectedTrack.times[0]);
+  plot.plot(selectedTrack, featureName, timeControls.getCurrentFrame());
+  await drawLoop();
```

Comment:
I'm not sure but drawloop is possibly redundant now as setCurrentFrame should cause a redraw too.

Reply:
Fixed, thanks!